util/db/repository.go

```diff
@@ -130,9 +138,20 @@ func (db *db) listRepositories(ctx context.Context, repoType *string) ([]*appsv1
 	var repos []*appsv1.Repository
 	for _, inRepo := range inRepos {
 		if repoType == nil || *repoType == inRepo.Type {
-			r, err := db.GetRepository(ctx, inRepo.URL)
+			r, err := db.getRepository(ctx, inRepo.URL)
```

Comment:
I have thought a little about the new call semantics for `getRepository()`  and the idea came to my mind that we might introduce a dedicated error type here instead for checking if `r != nil` to see whether it's a mis-configuration or not. That would make the semantics of `getRepository()` much more clear imho. What do you think?

Reply:
You certainly have a good point here. The semantic of the return value is not obvious from looking at the code. However, i still think it is a good idea to check for `r != nil`, since it is very unusual for Go methods to have an actual return while returning an error. Also introducing a new error type for an unexported helper function seems kind of over kill. Still, I get your point and I think you are right.

My suggestion is to rename the function to something like `tryGetRepository` or something. Is this something that you would agree with? If not, please let me know 🙂. Meanwhile I will thing of a good function name to make the intention clearer and make a code change proposal for you to review

Reply:
Sadly, I could not think of a better name then `tryGetRepository`. Maybe you have an idea? I also added some documentation.

If you still insist to add a custom error type, would you embedd the repository into the error for that case instead? Or would you just assume that if the error has a certain type a repository must be present?

Looking forward to your feedback! 🙂

Reply:
I'm not opinionated about the method name, as long as the behavior is clearly documented. You are right about that we don't need a custom error type for now, it would be a little over-engineering probably.

Reply:
Turns out, I implemented the custom error type after all :)